examples/basics/visuals/tube.py

```diff
@@ -0,0 +1,86 @@
+"""
+Demonstration of Tube
+"""
+
+import sys
+from vispy import app, scene
+from vispy.scene.visuals import Tube
+
+from colorsys import hsv_to_rgb
+import numpy as np
+
+def torus_knot(p=3, q=4, num=100):
+    '''
+    Returns points in the p, q torus knot.
+
+    Note that not all combinations of p and q give knots;
+    some curves may pass through themselves.
+    '''
+    data = np.zeros((num, 3), dtype=np.float64)
+    ts = np.linspace(0, 2*np.pi, num)
+    rs = np.cos(q*ts) + 2
+    data[:, 0] = rs*np.cos(p*ts)
+    data[:, 1] = rs*np.sin(p*ts)
+    data[:, 2] = -1*np.sin(q*ts)
+    return data
+
+if __name__ == '__main__':
+    canvas = scene.SceneCanvas(keys='interactive')
+    canvas.view = canvas.central_widget.add_view()
+
+    points1 = torus_knot(5, 3)[:-1] * 5.
+    points1[:, 0] -= 20.
+    points1[:, 2] -= 15.
+
+    points2 = points1.copy()
+    points2[:, 2] += 30.
+
+    points3 = points1.copy()
+    points3[:, 0] += 41.
+    points3[:, 2] += 30
+
+    points4 = points1.copy()
+    points4[:, 0] += 41.
+
+    colors = np.linspace(0, 1, len(points1))
+    colors = np.array([hsv_to_rgb(c, 1, 1) for c in colors])
```

Comment:
use [this](https://github.com/vispy/vispy/blob/master/vispy/color/color_space.py#L78) instead?